src/shared/sol-conffile.c

```diff
@@ -262,19 +363,22 @@ sol_conffile_fill_vector(const char *path, const char *fallback_paths)
     char *full_path = NULL;
     char *include = NULL;
     char *include_fallbacks = NULL;
-    GKeyFile *keyfile = NULL;
+    char *config_file_contents = NULL;
+    struct sol_file_reader *file_reader;
     struct sol_ptr_vector pv;
+    struct sol_json_scanner json_scanner;
 
-    keyfile = sol_conffile_load_keyfile_from_paths
-            (path, fallback_paths, &full_path);
-    if (!keyfile)
+    config_file_contents = sol_conffile_load_json_from_paths(path, fallback_paths, &full_path, &file_reader);
+    if (!config_file_contents)
         return;
 
-    sol_conffile_get_keyfile_include_paths
-        (keyfile, &include, &include_fallbacks);
+    sol_json_scanner_init(&json_scanner, config_file_contents, strlen(config_file_contents));
```

Comment:
The file size is already known here inside `sol_file_reader`, there's no need to call `strlen()`.